demos/projects/ESPRESSIF/esp32/main/azure_iot_freertos_esp32_main.c

```diff
@@ -0,0 +1,280 @@
+#include "esp_sntp.h"
+#include "esp_log.h"
+#include "freertos/FreeRTOS.h"
+#include "freertos/task.h"
+#include "freertos/semphr.h"
+#include "nvs_flash.h"
+
+#define SNTP_SERVER_FQDN "pool.ntp.org"
+
+static const char *TAG = "sample_azureiot";
+
+/*-----------------------------------------------------------*/
+
+extern void vStartDemoTask( void );
+
+/*-----------------------------------------------------------*/
+
+void vLoggingPrintf( const char * pcFormat,
+                     ... )
+{
+    va_list arg;
+
+    va_start( arg, pcFormat );
+    vprintf( pcFormat, arg );
+    va_end( arg );
+}
+
+/*-----------------------------------------------------------*/
+
+uint64_t ullGetUnixTime( void )
+{
+    time_t now = time(NULL);
+
+    if (now == (time_t)(-1))
+    {
+        ESP_LOGE(TAG, "Failed obtaining current time.\r\n");
+    }
+
+    return now;
+}
+
+/*-----------------------------------------------------------*/
+
+static xSemaphoreHandle s_semph_get_ip_addrs;
+static esp_ip4_addr_t s_ip_addr;
+
+#define NR_OF_IP_ADDRESSES_TO_WAIT_FOR 1
+
+#if CONFIG_SAMPLE_IOT_WIFI_SCAN_METHOD_FAST
+#define SAMPLE_IOT_WIFI_SCAN_METHOD WIFI_FAST_SCAN
+#elif CONFIG_SAMPLE_IOT_WIFI_SCAN_METHOD_ALL_CHANNEL
+#define SAMPLE_IOT_WIFI_SCAN_METHOD WIFI_ALL_CHANNEL_SCAN
+#endif
+
+#if CONFIG_SAMPLE_IOT_WIFI_CONNECT_AP_BY_SIGNAL
+#define SAMPLE_IOT_WIFI_CONNECT_AP_SORT_METHOD WIFI_CONNECT_AP_BY_SIGNAL
+#elif CONFIG_SAMPLE_IOT_WIFI_CONNECT_AP_BY_SECURITY
+#define SAMPLE_IOT_WIFI_CONNECT_AP_SORT_METHOD WIFI_CONNECT_AP_BY_SECURITY
+#endif
+
+#if CONFIG_SAMPLE_IOT_WIFI_AUTH_OPEN
+#define SAMPLE_IOT_WIFI_SCAN_AUTH_MODE_THRESHOLD WIFI_AUTH_OPEN
+#elif CONFIG_SAMPLE_IOT_WIFI_AUTH_WEP
+#define SAMPLE_IOT_WIFI_SCAN_AUTH_MODE_THRESHOLD WIFI_AUTH_WEP
+#elif CONFIG_SAMPLE_IOT_WIFI_AUTH_WPA_PSK
+#define SAMPLE_IOT_WIFI_SCAN_AUTH_MODE_THRESHOLD WIFI_AUTH_WPA_PSK
+#elif CONFIG_SAMPLE_IOT_WIFI_AUTH_WPA2_PSK
+#define SAMPLE_IOT_WIFI_SCAN_AUTH_MODE_THRESHOLD WIFI_AUTH_WPA2_PSK
+#elif CONFIG_SAMPLE_IOT_WIFI_AUTH_WPA_WPA2_PSK
+#define SAMPLE_IOT_WIFI_SCAN_AUTH_MODE_THRESHOLD WIFI_AUTH_WPA_WPA2_PSK
+#elif CONFIG_SAMPLE_IOT_WIFI_AUTH_WPA2_ENTERPRISE
+#define SAMPLE_IOT_WIFI_SCAN_AUTH_MODE_THRESHOLD WIFI_AUTH_WPA2_ENTERPRISE
+#elif CONFIG_SAMPLE_IOT_WIFI_AUTH_WPA3_PSK
+#define SAMPLE_IOT_WIFI_SCAN_AUTH_MODE_THRESHOLD WIFI_AUTH_WPA3_PSK
+#elif CONFIG_SAMPLE_IOT_WIFI_AUTH_WPA2_WPA3_PSK
+#define SAMPLE_IOT_WIFI_SCAN_AUTH_MODE_THRESHOLD WIFI_AUTH_WPA2_WPA3_PSK
+#elif CONFIG_SAMPLE_IOT_WIFI_AUTH_WAPI_PSK
+#define SAMPLE_IOT_WIFI_SCAN_AUTH_MODE_THRESHOLD WIFI_AUTH_WAPI_PSK
+#endif
+
+/**
+ * @brief Checks the netif description if it contains specified prefix.
+ * All netifs created withing common connect component are prefixed with the module TAG,
+ * so it returns true if the specified netif is owned by this module
+ */
+static bool is_our_netif(const char *prefix, esp_netif_t *netif)
+{
+    return strncmp(prefix, esp_netif_get_desc(netif), strlen(prefix) - 1) == 0;

```

Comment:
> - 1 [](http://example.com/codeflow?start=68&length=4)

is -1 required?

Reply:
I haven't debugged this code. At this point that's a P2 for me as this code came from esp-idf samples.

Reply:
But that's a good question though. I'll keep the comment open even after merging so I can follow up.